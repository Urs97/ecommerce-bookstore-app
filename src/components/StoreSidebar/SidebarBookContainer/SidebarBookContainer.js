import React from 'react';
import './SidebarBookContainer.scss';
import SidebarBook from './SidebarBook/SidebarBook';

const SidebarBookContainer = ({ title, bookData }) => {
    return (
        <section className="sidebar-book-container">
            <h4 className="sidebar-title">{title}</h4>
            <SidebarBook book={bookData[0]}/>
            <SidebarBook book={bookData[1]}/>
            <SidebarBook book={bookData[2]}/>
        </section>
       
    )
};

export default SidebarBookContainer;