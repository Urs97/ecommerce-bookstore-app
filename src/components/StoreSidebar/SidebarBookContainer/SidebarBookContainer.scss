@import '../../../styles/shared.scss';

.sidebar-book-container > * {
    margin-bottom: $vertical-sidebar-padding;
}