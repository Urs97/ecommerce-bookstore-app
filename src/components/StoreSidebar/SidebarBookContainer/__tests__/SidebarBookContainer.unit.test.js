import React from 'react';
import {shallow} from 'enzyme';
import SidebarBookContainer from '../SidebarBookContainer';

describe('SidebarBookContainer', () => {
    test('renders empty SidebarBookContainer', () => {
        const wrapper = shallow(
            <SidebarBookContainer />
        );
        expect(wrapper).toMatchSnapshot();
    });
});