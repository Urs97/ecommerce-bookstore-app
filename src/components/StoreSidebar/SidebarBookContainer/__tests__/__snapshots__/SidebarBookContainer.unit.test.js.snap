// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarBookContainer renders SidebarBookContainer 1`] = `
<section
  className="sidebar-book-container"
>
  <h4
    className="sidebar-title"
  >
    Top Rated Books
  </h4>
  <SidebarBook
    book={
      Object {
        "authors": Array [
          Object {
            "key": "/authors/OL25342A",
            "name": "Marry Shelley",
          },
        ],
        "cover_edition_key": "OL6147070M",
        "key": "/works/OL450125W",
        "price": Array [
          94,
          99,
        ],
        "title": "Frankenstein",
      }
    }
  />
  <SidebarBook
    book={
      Object {
        "authors": Array [
          Object {
            "key": "/authors/OL13066A",
            "name": "H. G. Wells",
          },
        ],
        "cover_edition_key": "OL8000344M",
        "key": "/works/OL52267W",
        "price": Array [
          75,
          65,
        ],
        "title": "The Time Machine",
      }
    }
  />
  <SidebarBook
    book={
      Object {
        "authors": Array [
          Object {
            "key": "/authors/OL113611A",
            "name": "Jules Verne",
          },
        ],
        "cover_edition_key": "OL7027686M",
        "key": "/works/OL1099513W",
        "price": Array [
          54,
          38,
        ],
        "title": "Journey to the Center of the Earth",
      }
    }
  />
</section>
`;
